core/src/components/button/button.ionic.scss

```diff
@@ -109,7 +109,18 @@
 ::slotted(ion-icon[slot="start"]),
 ::slotted(ion-icon[slot="end"]),
 ::slotted(ion-icon[slot="icon-only"]) {
-  font-size: 1em;
+  font-size: px-to-rem(16);
+}
+
+:host(.button-xsmall),
+:host(.button-small),
+:host(.button-large),
+:host(.button-xlarge) {
+  ::slotted(ion-icon[slot="start"]),
+  ::slotted(ion-icon[slot="end"]),
+  ::slotted(ion-icon[slot="icon-only"]) {
+    font-size: 1em;
+  }
```

Comment:
What are the differences between these two groups? It seems that the second grouping isn't necessary since the first one is able to convert the px to rem as intended.

Reply:
The first one will set the font-size with 16px converted to **rem**, the second one will be 1**em**.
The first one is a request made by design for the default button size, the second one is also a request based on button size.
**rem** unit is based on browser default size and **em** unit is based on context size.
That's why they need be be like that ;)